adapter/src/debug_session/variables.rs

```diff
@@ -224,12 +224,21 @@ impl super::DebugSession {
 
         let mem_ref = match self.client_caps.supports_memory_references {
             Some(true) => {
-                let load_addr = var.load_address();
-                match load_addr {
-                    lldb::INVALID_ADDRESS => None,
-                    _ => Some(format!("0x{:X}", load_addr)),
+              match var.value_type() {
+                // Assume that the register value is an address, and use that.
+                // So users can dump memory, e.g. as the SP, RIP+offset, etc.
+                ValueType::Register => Some(format!("0x{}", value)),
```

Comment:
I guess this is a hack. Actually, the 'value' here is the "string" value of the register returned by lldb SBValue api, which happens to already be in hex.

I think the right thing to do it get the value as unsigned, and format it per above. I'll do that. Since we know it's a register, we can always assume it is representable as some sort of unsigned value (and assume that 64 bits is enough for anyone).

Reply:
done.